src/utils/usedAndSelectedResources.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtTreeItem } from "@microsoft/vscode-azext-utils";
+import { TreeView } from "vscode";
+import { ext } from "../extensionVariables";
+
+
+export async function getSelectedAzureResource(): Promise<string | undefined> {
+    const resourceTreeView = (ext.appResourceTreeView as TreeView<AzExtTreeItem>);
```

Comment:
I don't think these trees actually use `AzExtTreeItem`, right @alexweininger?

Reply:
I went off the fact that this is the assignment to the extension variable.

`ext.appResourceTreeView = treeView as unknown as vscode.TreeView<AzExtTreeItem>;`

Reply:
I would do `{ id?: string }`.